### PR TITLE
use file hash in mlbf key; drop salt

### LINKED ISSUE #13714: update bloomfilter key generation to use xpi file hash
> if you include a hash of the signed add-on in the murmer computation, you should be okay as long as you're producing a 256-bit random value per signature computation. To clarify, would this now constitute your murmer computation: murmerHash3(ID + version + sha_256(file+sig_on_file)) ?

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #13714

### CODE REVIEW

File: src/olympia/blocklist/mlbf.py [hunk 1]
```diff
@@ -25,28 +25,33 @@ def get_blocked_guids():
...
-    return Version.unfiltered.values_list('addon__guid', 'version')
+    return File.objects.values_list(
+        'version__addon__guid', 'version__version', 'hash')
 
 
 def hash_filter_inputs(input_list, key_format):
     return [
-        key_format.format(guid=guid, version=version)
-        for (guid, version) in input_list]
+        key_format.format(guid=guid, version=version, xpi_hash=hash)
```
Comment (willdurand) [inline src/olympia/blocklist/mlbf.py]: Given this line, I'd expect `hash` to be `xpi_hash` everywhere in this patch. That would make it slightly more future-proof as the `key_format` expects `xpi_hash` so potentially a further iteration might introduce `hash` a placeholder (maybe not but we never know and that would still be more consistent)

File: src/olympia/blocklist/tests/test_mlbf.py [hunk 2]
```diff
@@ -41,65 +41,70 @@ def setUp(self):
...
         assert len(all_guids) == File.objects.count() == 10 + 8
-        assert (self.three_ver.guid, '123.40') in all_guids
-        assert (self.three_ver.guid, '123.5') in all_guids
-        assert (self.three_ver.guid, '123.45.1') in all_guids
-        over_tuple = (self.over.guid, self.over.addon.current_version.version)
-        under_tuple = (
-            self.under.guid, self.under.addon.current_version.version)
-        assert over_tuple in all_guids
-        assert under_tuple in all_guids
+        assert (self.three_ver.guid, '123.40', '') in all_guids
```
Comment (willdurand) [inline src/olympia/blocklist/tests/test_mlbf.py]: Why is there no xpi hash in these `assert`?

File: src/olympia/blocklist/tests/test_mlbf.py [hunk 3]
```diff
@@ -41,65 +41,70 @@ def setUp(self):
...
-        assert (self.three_ver.guid, '123.5') in blocked_guids
-        assert (self.three_ver.guid, '123.45.1') not in blocked_guids
-        over_tuple = (self.over.guid, self.over.addon.current_version.version)
-        under_tuple = (
-            self.under.guid, self.under.addon.current_version.version)
-        assert over_tuple not in blocked_guids
-        assert under_tuple not in blocked_guids
+        assert (self.three_ver.guid, '123.40', '') in blocked_guids
+        assert (self.three_ver.guid, '123.5', '') in blocked_guids
+        assert (self.three_ver.guid, '123.45.1', '') not in blocked_guids
```
Comment (willdurand) [inline src/olympia/blocklist/tests/test_mlbf.py]: Same question here.

File: src/olympia/blocklist/mlbf.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/blocklist/mlbf.py]: The property is `File.hash` and, imo, just `hash` is ambiguous for Firefox so I went with `xpi_hash`.  So the mismatch is going to happen somewhere.

File: src/olympia/blocklist/tests/test_mlbf.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/blocklist/tests/test_mlbf.py]: hash wasn't set when the addon/version was created so it's the default (`''`).

File: src/olympia/blocklist/tests/test_mlbf.py (hunk 3 above)
Comment (eviljeff) [reply src/olympia/blocklist/tests/test_mlbf.py]: hash wasn't set when the addon/version was created so it's the default (`''`).

File: src/olympia/blocklist/mlbf.py (hunk 1 above)
Comment (willdurand) [reply src/olympia/blocklist/mlbf.py]: I see. Would it be worth changing here, though?

File: src/olympia/blocklist/tests/test_mlbf.py (hunk 2 above)
Comment (willdurand) [reply src/olympia/blocklist/tests/test_mlbf.py]: Can we set more realistic values? Or do we have non-default values in other test cases? (I quickly looked but could not find anything, maybe I missed something)

File: src/olympia/blocklist/mlbf.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/blocklist/mlbf.py]: which change are you suggesting exactly?

Comment (willdurand) [reply src/olympia/blocklist/mlbf.py]: Sorry, renaming `hash` to `xpi_hash` in this file (even if the `File` property is named `hash`).

**Comment by diox** [comment]
Will likely have to be dropped as including the file hash is easy for us but not for Firefox. We are looking into going back to the salt approach but with a different hash.

**Comment by eviljeff** [comment]
closing this for now as it's not a good change on the Firefox side as @diox said

**Pull request closed without merging**